geth/node/rpc.go

```diff
@@ -94,6 +110,66 @@ func (c *RPCManager) Call(inputJSON string) string {
 	return c.makeJSONErrorResponse(ErrRPCServerTimeout)
 }
 
+// callNodeStream delivers giving request and body content to the external ethereum
+// (infura) RPCServer to process the request and returns response.
+func (c *RPCManager) callUpstreamStream(config *params.NodeConfig, body io.Reader) ([]byte, error) {
+	upstreamURL, err := url.Parse(config.UpstreamConfig.URL)
+	if err != nil {
+		return nil, err
+	}
+
+	httpReq := httptest.NewRequest("POST", "/", body)
```

Comment:
I think my issue had something to do with the URL used during the test, causing the issue i faced. This should work now. Thanks for the catch.